### PR TITLE
DX-1672 Add path-based permission example for Operator

### PR DESCRIPTION
**Pull request by carolwkc**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

[DX-1672](https://tyktech.atlassian.net/browse/DX-1672)
---

### Preview Link

[preview](https://deploy-preview-5375--tyk-docs.netlify.app/docs/nightly/tyk-stack/tyk-operator/secure-an-api)

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have reviewed the suggestions made by our AI (PR Agent) and updated them accordingly (spelling errors, rephrasing, etc.)
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labeled** the PR

___

### **PR Type**
documentation

___

### **Description**
- Added a new example for path-based permissions in the Tyk Operator documentation.
- Introduced the `allowed_urls` field in the API specification to restrict access based on specific paths and HTTP methods.
- Updated the documentation to include explanations for the new path-based permissions feature.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>secure-an-api.md</strong><dd><code>Add path-based permissions example and documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-stack/tyk-operator/secure-an-api.md

<li>Added example for path-based permissions in API spec.<br> <li> Introduced <code>allowed_urls</code> field to restrict access by path and method.<br> <li> Updated documentation to explain path-based permissions.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5375/files#diff-62583238874aadf2fcc63ad539d89257c7bd38c3b7d35bf52a30015156f26dd4">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

[DX-1672]: https://tyktech.atlassian.net/browse/DX-1672?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tyk-docs/content/tyk-stack/tyk-operator/secure-an-api.md [hunk 1]
```diff
@@ -142,6 +146,10 @@ Access lists for API and versions:
 
 - `access_right_array`: The list of APIs security policy has access to.
 
+Path-based permissions for API:
+
+- `allowed_urls`: Restrict access on a per-path and per method basis to only allow access to specific portions of the API
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-operator/secure-an-api.md]:
```suggestion
- `allowed_urls`: Restrict access per path and per method to restrict access to specific portions of the API
```

**Review by dcs3spp — APPROVED** [review]
Thanks @caroltyk PR LGTM once peer approved we will merge and release

File: tyk-docs/content/tyk-stack/tyk-operator/secure-an-api.md [hunk 2]
```diff
@@ -142,6 +146,10 @@ Access lists for API and versions:
 
 - `access_right_array`: The list of APIs security policy has access to.
 
+Path-based permissions for API:
+
+- `allowed_urls`: Restrict access per path and per method to restrict access to specific portions of the API
```
Comment (komalsukhani) [inline tyk-docs/content/tyk-stack/tyk-operator/secure-an-api.md]:
```suggestion
- `allowed_urls`: Restrict access per path and per method to specific portions of the API
```

**Comment by dcs3spp** [comment]
@komalsukhani Can you approve if ok and I will merge and release

**Pull request merged**